### PR TITLE
Big-endian test case fixes: Xoshiro hash

### PR DESCRIPTION
**Pull request by uweigand**
* Skip endian-dependent part of Xoshiro_AlgorithmBehavesAsExpected on
  big-endian systems

### CODE REVIEW

File: src/libraries/System.Runtime.Extensions/tests/System/Random.cs [hunk 1]
```diff
@@ -414,6 +414,12 @@ public void Xoshiro_AlgorithmBehavesAsExpected()
             Assert.NotNull(s2);
             Assert.NotNull(s3);
 
+            if (!BitConverter.IsLittleEndian)
+            {
+                // The remainder of the test expects specific result values that will
+                // only happen on little-endian platforms.
+                return;
+            }
```
Comment (stephentoub) [inline src/libraries/System.Runtime.Extensions/tests/System/Random.cs:422]: The stuff above this isn't meaningful to test.  We should just disable the whole test on big-endian.

Comment (akoeplinger) [reply src/libraries/System.Runtime.Extensions/tests/System/Random.cs:422]: I agree. You can do that by adding the following attribute to the test:

```csharp
[ConditionalFact(typeof(BitConverter), nameof(BitConverter.IsLittleEndian))]
```

Comment (uweigand) [reply src/libraries/System.Runtime.Extensions/tests/System/Random.cs:422]: @akoeplinger This doesn't appear to work for me, the test case still fails and I get:
`System.InvalidOperationException : An appropriate member 'IsLittleEndian' could not be found. The conditional method needs to be a static method or property on the type System.BitConverter or any ancestor, of any visibility, accepting zero arguments, and having a return type of Boolean.`

Comment (stephentoub) [reply src/libraries/System.Runtime.Extensions/tests/System/Random.cs:422]: We should update the attribute discoverers to allow fields as well.

Comment (uweigand) [reply src/libraries/System.Runtime.Extensions/tests/System/Random.cs]: I've seen other tests use a local property, so I've tried this and it seems to work.  Patch updated accordingly.   Does this look OK?

Comment (stephentoub) [reply src/libraries/System.Runtime.Extensions/tests/System/Random.cs]: > We should update the attribute discoverers to allow fields as well.

https://github.com/dotnet/arcade/pull/7117

**Comment by uweigand** [comment]
@stephentoub Looks like our last two updates crossed each other, sorry.   Thanks for the quick Arcade fix!  However, if I understand things correctly, this new feature won't be available to "runtime" until there's a new release of Arcade (the Microsoft.DotNet.XUnitExtensions package) and that new release is then pulled into runtime, right?

Should we just wait until that happens, or would my latest version of this patch using the intermediate local property be acceptable to merge as interim solution:
```
        public static bool IsLittleEndian => BitConverter.IsLittleEndian;
        [ConditionalFact(nameof(IsLittleEndian))]
```

**Comment by danmoseley** [comment]
You should be able to just wait for https://github.com/dotnet/runtime/pull/49820

File: src/libraries/System.Runtime.Extensions/tests/System/Random.cs [hunk 2]
```diff
@@ -388,7 +388,10 @@ public void Empty_Success(bool derived, bool seeded)
             r.NextBytes(Span<byte>.Empty);
         }
 
-        [Fact]
+        // The following test makes assumptions about the implementation that hold
+        // true only on little-endian platforms.
+        public static bool IsLittleEndian => BitConverter.IsLittleEndian;
+        [ConditionalFact(nameof(IsLittleEndian))]
```
Comment (stephentoub) [inline src/libraries/System.Runtime.Extensions/tests/System/Random.cs:394]:
```suggestion
        [ConditionalFact(typeof(BitConverter), nameof(BitConverter.IsLittleEndian))] // test makes little-endian assumptions
```

**Review by akoeplinger — APPROVED** [review]

**Comment by uweigand** [comment]
This works as expected on my big-endian machine, thanks!

**Pull request merged**